hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -170,6 +171,15 @@ public static IOException translateException(@Nullable String operation,
         operation,
         StringUtils.isNotEmpty(path)? (" on " + path) : "",
         exception);
+
+    // Exceptions from encryption client are wrapped in S3EncryptionClientException, so unwrap.
+    if (exception instanceof S3EncryptionClientException) {
```

Comment:
this will make the new library mandatory.  need some more work here, such as simply looking a class details of nested exception. should also go into `org.apache.hadoop.fs.s3a.impl.ErrorTranslation`
with a unit test